backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/VersionController.kt

```diff
@@ -1,20 +1,47 @@
 package com.tngtech.apicenter.backend.connector.rest.controller
 
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper
+
+import com.tngtech.apicenter.backend.domain.entity.Version
 import com.tngtech.apicenter.backend.connector.rest.dto.VersionDto
 import com.tngtech.apicenter.backend.connector.rest.mapper.VersionDtoMapper
 import com.tngtech.apicenter.backend.domain.handler.VersionHandler
 import org.springframework.web.bind.annotation.DeleteMapping
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestHeader
 import org.springframework.web.bind.annotation.RestController
 import java.util.UUID
+import javax.servlet.http.HttpServletResponse
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
 
 @RestController
 class VersionController constructor(private val versionHandler: VersionHandler, private val versionDtoMapper: VersionDtoMapper) {
 
-    @GetMapping("/specifications/{specificationId}/versions/{version}")
-    fun findVersion(@PathVariable specificationId: UUID, @PathVariable version: String): VersionDto = versionDtoMapper.fromDomain(versionHandler.findOne(specificationId, version))
+    @RequestMapping("/specifications/{specificationId}/versions/{version}",
+        produces = ["application/json", "application/yml"],
+        headers = ["Accept=application/json", "Accept=application/yml"],
+        method = [RequestMethod.GET])
+    fun findVersion(@PathVariable specificationId: UUID, @PathVariable version: String, @RequestHeader("Accept") accept: String): VersionDto {
+        val foundVersion = versionHandler.findOne(specificationId, version)
+        if (accept == "application/yml") {
+            logger.info("YAML requested")
+            val jsonNodeTree: JsonNode = ObjectMapper().readTree(foundVersion.content)
+            val jsonAsYaml: String = YAMLMapper().writeValueAsString(jsonNodeTree)
+            logger.info(jsonAsYaml)
+            return versionDtoMapper.fromDomain(Version(foundVersion.version, jsonAsYaml))
+        } else {
```

Comment:
We have two cases: yml or json. Here you explicitly handle _everything but yml_ as the default case. Maybe we could explicitly handle the case when json is uploaded and move some sort of error handling in the else-part.

Reply:
In the current configuration, if the user doesn't specify a file format, JSON is returned. So this made sense to me? Perhaps I haven't understood your comment 😅

Reply:
I'm not sure if it makes sense to do the conversion here or somewhere else. What do you say?